server/src/main/java/org/elasticsearch/index/reindex/BulkByScrollTask.java

```diff
@@ -407,6 +529,114 @@ public XContentBuilder innerXContent(XContentBuilder builder, Params params)
             return builder;
         }
 
+        public static Status fromXContent(XContentParser parser) throws IOException {
+            XContentParser.Token token;
+            if (parser.currentToken() == Token.START_OBJECT) {
+                 token = parser.nextToken();
+            } else {
+                token = parser.nextToken();
+            }
+            ensureExpectedToken(Token.START_OBJECT, token, parser::getTokenLocation);
+            token = parser.nextToken();
+            ensureExpectedToken(Token.FIELD_NAME, token, parser::getTokenLocation);
+            return innerFromXContent(parser);
+        }
+
+        public static Status innerFromXContent(XContentParser parser) throws IOException {
+            Token token = parser.currentToken();
```

Comment:
I wonder if you can avoid all of this with a builder object that knows how to build either status or error. Then you can use regular ObjectParser with it.

Reply:
I believe I cannot build the error myself (as we already discussed)? I need to peek at a field name to realise what I need to parse which doesn't work well with an `ObjectParser`. I could declare all fields in the `ObjectParser` and then try to build an object using that but for that, I would need to write the parse for the error as well. Which I really do not want to do... Or are you suggesting something else?

Reply:
Could you add a comment here explaining why you need to parse this manually? I'm 100% sure when I read this again in 6 months I'll waste an hour figuring out why you did it this way. A comment will save future me some time.